web3.js/src/connection.ts

```diff
@@ -3023,104 +3023,168 @@ export class Connection {
 
     assert(decodedSignature.length === 64, 'signature has invalid length');
 
-    const subscriptionCommitment = commitment || this.commitment;
-    let timeoutId;
-    let subscriptionId;
-    let done = false;
+    if (
+      Object.prototype.hasOwnProperty.call(strategy, 'lastValidBlockHeight')
+    ) {
+      return await this.confirmTransactionUsingBlockheightExceedenceStrategy({
+        commitment: commitment || this.commitment,
+        strategy: strategy as BlockheightBasedTransactionConfirmationStrategy,
+        signature: rawSignature,
+      });
+    } else {
+      return await this.confirmTransactionUsingLegacyTimeoutStrategy({
+        commitment: commitment || this.commitment,
+        signature: rawSignature,
+      });
+    }
+  }
 
+  private getTransactionConfirmationPromise({
+    commitment,
+    signature,
+  }: {
+    commitment?: Commitment;
+    signature: string;
+  }): [
```

Comment:
nit: how about returning an object with named fields here that indicate what these return values are